### PR TITLE
diag_integral output includes ensemble number

### LINKED ISSUE #111: Adding ensemble number to diag_integral.out
A user requested adding the ensemble number to the end of the diag_integral.out filename to distinguish between ensembles.

> As background, the reason I'm interested in this is that I've used this 
file to figure out the radiative imbalance of the model in the past with 
single member runs. Now that I have ensembles to play with I'd like to 
be able to do the same thing for each ensemble member and see the spread.
I still get a single diag_integral.out file from my runs, but I have no 
idea which member the data belongs to or whether the data gets written 
by the first, last, or "last to get to the routine" member. I don't even 
know if different timesteps are from the same member or an average of 
the entire ensemble.

### PR DESCRIPTION
**Pull request by thomas-robinson**
Resolves #111
When an ensemble is run, the diag_integral.out file will be named diag_integral.ens_##.out where ## is a 2-digit number designated to the ensemble number.  This also resolves autotools compilation errors.

### CODE REVIEW

**Comment by thomas-robinson** [comment]
@wrongkindofdoctor you can squash the commits because there are a bunch of small ones where I try to get autotools to work.

File: diag_integral/diag_integral.F90 [hunk 1]
```diff
@@ -1522,7 +1527,40 @@ function vert_diag_integral (data, wt) result (data2)
...
 
-
+function ensemble_file_name(fname) result(updated_file_name)
+     character (len=mxch), intent(inout) :: fname
+     character (len=mxch) :: updated_file_name
+     integer :: ensemble_id_int
+     character(len=7) :: ensemble_suffix
+     character(len=2) :: ensemble_id_char 
+     integer :: i
+     !> Make sure the file name short enough ti handle adding the ensemble number
```
Comment (wrongkindofdoctor) [inline diag_integral/diag_integral.F90]: minor typo: 'ti' should be 'to'

File: diag_integral/diag_integral.F90 [hunk 2]
```diff
@@ -1522,7 +1527,40 @@ function vert_diag_integral (data, wt) result (data2)
...
+          ensemble_id_int = get_ensemble_id()
+          write(ensemble_id_char,"(I0)") ensemble_id_int
+     !> Add a 0 if the ensemble is less than 10 (2 digit)
+          if (ensemble_id_int < 10) then
+               ensemble_suffix = ".ens_0"//trim(ensemble_id_char)
+          elseif (ensemble_id_int >= 10 .and. ensemble_id_int < 100) then
+               ensemble_suffix = ".ens_"//trim(ensemble_id_char)
+          else
+               call error_mesg ('diag_integral_mod',  &
+                ' Does not support ensemble sizes over 99.', FATAL)
```
Comment (wrongkindofdoctor) [inline diag_integral/diag_integral.F90]: I know MAX_ENSEMBLE_SIZE is[ currently set to 100](https://github.com/NOAA-GFDL/FMS/blob/bb4e81fe3598af0069eb0ad576d6dd956275d209/coupler/ensemble_manager.F90#L35) in ensemble_manager.F90, but is this size sufficient to support the current (or anticipated) standard ensemble sizes defined in the atmos/ESM workflows of the users who want this feature implemented?

File: diag_integral/diag_integral.F90 (hunk 1 above)
Comment (thomas-robinson) [reply diag_integral/diag_integral.F90]: I am the master of the typo.  It is fixed now.

File: diag_integral/diag_integral.F90 (hunk 2 above)
Comment (thomas-robinson) [reply diag_integral/diag_integral.F90]: No, but this is a conversation that is beyond the scope of this merge request.  It's also a simple enough fix here, BUT running 100 model runs with 864 cores is also beyond the limits of what we are currently allowed to run on our production system.  This is definitely a larger conversation.

Comment (wrongkindofdoctor) [reply diag_integral/diag_integral.F90]: All right. Thanks for confirming that this is a resource limitation rather than a standard workload size.

**Comment by thomas-robinson** [comment]
`mxch` is the length of `file_name` which is the variable passed into the new routine.  `file_name` is a namelist variable, and I've only seen it set to "diag_integral.out".  In my first pass at this, I actually had an if statement that checked if `file_name` was set to diag_integral.out.  

Adding the ensemble name only adds 7 characters ".ens_00", so 64 should be sufficient.

**Pull request merged**